src/meson.build

```diff
@@ -560,7 +560,7 @@ defs_list = [harfbuzz_def]
 version = '0.@0@.0'.format(hb_version_int)
 
 extra_hb_cpp_args = []
-if cpp.get_argument_syntax() == 'msvc'
+if cpp.get_define('_MSC_FULL_VER') != ''
```

Comment:
Can't `cpp.get_id()` be used instead? Or is it that all 3 compilers shipped by Microsoft will result in `_MSC_FULL_VER` being defined?

Reply:
Using `cpp.get_id()` or the more precise `cpp.get_argument_syntax()` doesn't cover the case of Microsoft Clang (with GNU command line, as CMake would call it). However, all three do define the `_MSC_FULL_VER` macro that targets the version of Visual Studio that they ship with.